src/svm/interface/svm.c

```diff
@@ -1441,6 +1441,52 @@ PetscErrorCode SVMView(SVM svm,PetscViewer v)
   PetscFunctionReturn(0);
 }
 
+/*@
+  SVMViewTrainingPredictions - Views predictions on training samples using trained model.
+
+  Input Parameters:
++ svm - SVM context
+- v - visualization context
+
+  Level: beginner
+
+.seealso SVMLoadTrainingDataset, SVMPredict, PetscViewer
+@*/
+PetscErrorCode SVMViewTrainingPredictions(SVM svm,PetscViewer v)
+{
+
+  PetscFunctionBeginI;
```

Comment:
```suggestion
  PetscFunctionBegin;
```
No point tracing these

Reply:
I partially agree with you. There is no point in tracing this when you use PermonSVM on the local machine. However, when a problem is occurred with writing into a file on a remote machine, it is good to know that the program is stuck there. From this point of view, tracing makes sense.

Reply:
You should use PETSc's -log_trace (because of granularity) or proper debugging tools if your program is getting stuck. PERMON's trace is a bit overused in the main library as well - it should trace only really high-level functions like QPT* and QPSSolve.